.github/workflows/check-lurk-compiles.yml

```diff
@@ -21,26 +18,29 @@ jobs:
           repository: lurk-lab/ci-workflows
       - uses: ./.github/actions/ci-env
       - uses: actions/checkout@v4
-        with:
-          path: ${{ github.workspace }}/${{ github.repository }}
       - uses: actions/checkout@v4
         with:
           repository: lurk-lab/lurk-rs
-          path: ${{ github.workspace }}/lurk
+          path: ./lurk-rs
           submodules: recursive
       - uses: dtolnay/rust-toolchain@stable
       - uses: Swatinem/rust-cache@v2
       - name: Patch Cargo.toml
-        working-directory: ${{ github.workspace }}/lurk
+        working-directory: ${{ github.workspace }}/lurk-rs
         run: |
-          REPOSITORY_NAME=$(echo ${{ github.repository }} | awk -F'/' '{ print $2 }')
-          echo "[patch.'https://github.com/${{ github.repository }}']" >> Cargo.toml
-          if [ ! -z "${{ inputs.package-name }}" ];
+          URL=https://github.com/${{ github.repository }}
+          # the dependency we want to patch is usually the same as the package, but
+          # we e.g. want to override dependency 'nova' with an 'arecibo' package
+          DEPENDENCY=$(grep "git = \"$URL\"" Cargo.toml | awk '{ print $1 }')
```

Comment:
oooh, that's clever. It won't work for bellpepper, where you have two lines to replace ($DEPENDENCY in {bellpepper, bellpepper-core}), but it will work for the simple case.